src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -2124,9 +2154,9 @@ class BatchSegment(BaseSegment):
     match_grammar = OneOf(
         # Things that can be bundled
         AnyNumberOf(
-            OneOf(
-                Ref("BeginEndSegment"),
+            Sequence(
                 Ref("StatementSegment"),
```

Comment:
Sure.  Since introduction of the BatchSegment, StatementSegments are commands that can be bundled together 1-X times within a batch.  A CREATE PROCEDURE statement is different in TSQL in that it must be the only statement within a batch.  I believe there are other statements which behave similarly but I haven't yet looked for a comprehensive list.

Reply:
makes sense, thanks for clarifying 😄